src/app/components/user-details/user-details.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { UserDetails, UserUpdate } from 'src/app/models/user';
+import { UserService } from 'src/app/services/user.service';
+
+@Component({
+    selector: 'app-user-details',
+    templateUrl: './user-details.component.html',
+    styleUrls: ['./user-details.component.scss']
+})
+export class UserDetailsComponent implements OnInit {
+    loading: boolean = false;
+    editMode: boolean = false;
+
+    userEditForm!: FormGroup;
+    email: FormControl;
+
+    tempDetails = {
```

Comment:
Instead of having this, you should have an actual 'user' object where you save the result from the getUserById() call.